silverfin/silverfin-cli
# Support Staging environments

## Description

Fixes #131 

For Silverfin Staging, Basic authentication needs to be added in the request header, with the proper credentials.

## Updates 

- New commands to set and get HOST
- Maintain also HOST as ENV variable for compatibility
- Support Staging Environments
- Log stored firms and host before each command
- Extract into dedicated class and add tests

## How to test

You need to create a new ENV variable named `SF_BASIC_AUTH`  (you can reach out to me in case you don't have it yet)

Different commands that should be tested.
NOTE: when you switch hosts (for example, from live to staging) a new `silverfin authorize` would need to be run first.
TIP: always use `--verbose` flag in all commands

```bash
silverfin authorize
silverfin config --get-host
silverfin config --set-host # switch between live and staging
silverfin import-reconciliation # from live/staging and firm/partners
silverfin config --refresh-token
silverfin config --refresh-partner-token
```

For example, some steps that can be followed:

```bash
silverfin --set-host <staging-url>
silverfin authorize
silverfin import-reconciliation # from firm
silverfin import-reconciliation # from partners
silverfin config --get-host
silverfin config --refresh-token
silverfin config --refresh-partner-token
silverfin --set-host <production-url>
silverfin authorize
silverfin import-reconciliation # from firm
silverfin import-reconciliation # from partners
silverfin config --refresh-token
silverfin config --refresh-partner-token
```

## Type of change


## Checklist

- [X] Version updated (if needed)
